entity-service/src/test/kotlin/com/egm/stellio/entity/service/EntityServiceTests.kt

```diff
@@ -826,6 +820,190 @@ class EntityServiceTests {
+        every { mockkedEntity.properties } returns mutableListOf()
+
+        every { neo4jRepository.hasPropertyInstance(any(), any()) } returns false
+        every { entityRepository.findById(any()) } returns Optional.of(mockkedEntity)
+        every { mockkedEntity.id } returns entityId
+        every { neo4jRepository.createPropertyOfSubject(any(), any()) } returns UUID.randomUUID().toString()
+
+        entityService.appendEntityAttributes(entityId, expandedNewProperty, false)
+
+        verify {
+            neo4jRepository.hasPropertyInstance(
+                match {
+                    it.id == entityId &&
+                        it.label == "Entity"
+                },
+                "https://ontology.eglobalmark.com/aquac#fishNumber",
+                datasetId = URI.create("urn:ngsi-ld:Dataset:fishNumber:1")
+            )
+        }
+        verify {
+            neo4jRepository.hasPropertyInstance(
+                match {
+                    it.id == entityId &&
+                        it.label == "Entity"
+                },
+                "https://ontology.eglobalmark.com/aquac#fishNumber",
+                datasetId = null
+            )
+        }
+
+        verify { entityRepository.findById(eq(entityId)) }
+        verify { neo4jRepository.createPropertyOfSubject(
+            match {
+                it.id == entityId &&
+                    it.label == "Entity"
+            },
+            match {
+                it.value == 500 &&
+                it.name == "https://ontology.eglobalmark.com/aquac#fishNumber" &&
+                it.datasetId == URI.create("urn:ngsi-ld:Dataset:fishNumber:1")
+            })
+        }
+        verify { neo4jRepository.createPropertyOfSubject(
+            match {
+                it.id == entityId &&
+                    it.label == "Entity"
+            },
+            match {
+                it.value == 600 &&
+                    it.name == "https://ontology.eglobalmark.com/aquac#fishNumber" &&
+                it.datasetId == null
+            })
+        }
+        verify { neo4jRepository.updateEntityModifiedDate(eq(entityId)) }
+
+        confirmVerified()
+    }
+
+    @Test
+    fun `it should not override the default instance if overwrite is disallowed`() {
+
+        val entityId = "urn:ngsi-ld:Beehive:123456"
+        val newProperty = """
+            {
+              "fishNumber": [{
+                "type": "Property",
+                "value": 600
+              }]
+            }
+        """.trimIndent()
+        val expandedNewProperty =
+            NgsiLdParsingUtils.expandJsonLdFragment(
+                newProperty,
+                aquacContext!!
+            )
+
+        val mockkedEntity = mockkClass(Entity::class)
+
+        every { mockkedEntity.id } returns entityId
+        every { entityRepository.findById(any()) } returns Optional.of(mockkedEntity)
+
+        every { neo4jRepository.hasPropertyInstance(any(), any()) } returns false
```

Comment:
Is not that the inverse of what you are testing?

Reply:
so you should verify that the createPropertyOfSubject is not called.

here, you changed the returned value of the method from false to true and the test is still passing. that's an indicator that something is not well tested.